pkg/plugins/trigger/push.go

```diff
@@ -21,27 +21,44 @@ import (
 
 	prowapi "sigs.k8s.io/prow/pkg/apis/prowjobs/v1"
 	"sigs.k8s.io/prow/pkg/config"
+	"sigs.k8s.io/prow/pkg/git/v2"
 	"sigs.k8s.io/prow/pkg/github"
 	"sigs.k8s.io/prow/pkg/pjutil"
 )
 
-func listPushEventChanges(pe github.PushEvent) config.ChangedFilesProvider {
+func listPushEventChanges(gc git.ClientFactory, pe github.PushEvent) config.ChangedFilesProvider {
 	return func() ([]string, error) {
-		changed := make(map[string]bool)
+
+		// Fallback to use PushEvent
+		changes := make(map[string]bool)
 		for _, commit := range pe.Commits {
 			for _, added := range commit.Added {
-				changed[added] = true
+				changes[added] = true
 			}
 			for _, removed := range commit.Removed {
-				changed[removed] = true
+				changes[removed] = true
 			}
 			for _, modified := range commit.Modified {
-				changed[modified] = true
+				changes[modified] = true
 			}
 		}
 		var changedFiles []string
-		for file := range changed {
-			changedFiles = append(changedFiles, file)
+		if len(changes) == github.ChangesFilesLimit && gc != nil {
```

Comment:
As above for pull requests, only load the new code if exactly 3000 files changes are found.